src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/ReasonableAdjustmentsMapper.java

```diff
@@ -0,0 +1,185 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.EventRequestData;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+import uk.gov.hmcts.reform.prl.clients.ccd.CcdCoreCaseDataService;
+import uk.gov.hmcts.reform.prl.enums.CaseEvent;
+import uk.gov.hmcts.reform.prl.enums.PartyEnum;
+import uk.gov.hmcts.reform.prl.models.Element;
+import uk.gov.hmcts.reform.prl.models.c100rebuild.C100RebuildApplicantDetailsElements;
+import uk.gov.hmcts.reform.prl.models.caseflags.Flags;
+import uk.gov.hmcts.reform.prl.models.caseflags.flagdetails.FlagDetail;
+import uk.gov.hmcts.reform.prl.models.caseflags.request.C100FlagDetailRequest;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+import uk.gov.hmcts.reform.prl.services.citizen.CaseService;
+import uk.gov.hmcts.reform.prl.utils.CaseUtils;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.prl.utils.ElementUtils.element;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ReasonableAdjustmentsMapper {
+    private final IdamClient idamClient;
+    private final CcdCoreCaseDataService coreCaseDataService;
+    @Autowired
+    private final ObjectMapper objectMapper;
+    private final CaseService caseService;
+
+    public CaseData mapRAforC100MainApplicant(String applicants, CaseData caseData, String authToken) throws JsonProcessingException {
+        log.info("Inside mapReasonableAdjustmentsForC100MainApplicant applicants {}", applicants);
+        log.info("Inside mapReasonableAdjustmentsForC100MainApplicant caseData {}", caseData);
+
+        if (isNotEmpty(applicants)) {
+            String caseId = String.valueOf(caseData.getId());
+            C100RebuildApplicantDetailsElements applicantDetails = new ObjectMapper()
+                .readValue(
+                    applicants,
+                    C100RebuildApplicantDetailsElements.class
+                );
+
+            log.info("Inside mapReasonableAdjustmentsForC100MainApplicant applicantDetails {}", applicantDetails);
+
+            if (CollectionUtils.isNotEmpty(applicantDetails.getApplicants()) && CollectionUtils.isNotEmpty(
+                applicantDetails.getApplicants().get(0).getReasonableAdjustmentsFlags())) {
```

Comment:
null pointer check on get(0)

Reply:
I did not put that because I am checking if applicants are there before that which means get(0) is always present.